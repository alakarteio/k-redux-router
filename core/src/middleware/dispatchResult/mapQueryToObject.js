/* eslint-env browser */

// https://stackoverflow.com/questions/8648892/convert-url-parameters-to-a-javascript-object
export default () => {
  const { location } = window

  if (location.search.length < 2) return {}

  const search = location.search.substring(1)
  return JSON.parse(`{"${decodeURI(search).replace(/"/g, '\\"').replace(/&/g, '","').replace(/=/g, '":"')}"}`)
}
