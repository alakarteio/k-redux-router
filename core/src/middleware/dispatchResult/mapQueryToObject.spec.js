/* eslint-env jest */
import mapQueryToObject from './mapQueryToObject'

describe('mapQueryToObject', () => {
  const match = res => () => expect(res).toMatchSnapshot()

  it('should map the query param of a path to an object', match(mapQueryToObject({
    location: {
      search: '&should=be&an=object&with=4&parameters=true',
    },
  })))

  it('should returns an empty object when just the ?', match(mapQueryToObject({ location: { search: '?' } })))
  it('should returns an empty object when no query', match(mapQueryToObject({ location: { search: '' } })))
})
