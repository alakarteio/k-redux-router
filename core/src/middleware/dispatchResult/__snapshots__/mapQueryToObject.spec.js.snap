// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapQueryToObject should map the query param of a path to an object 1`] = `
Object {
  "an": "object",
  "parameters": "true",
  "should": "be",
  "with": "4",
}
`;

exports[`mapQueryToObject should returns an empty object when just the ? 1`] = `Object {}`;

exports[`mapQueryToObject should returns an empty object when no query 1`] = `Object {}`;
